src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -122,7 +122,8 @@ internal static string Platform
         {
             get
             {
-                return ("Windows NT");
+                OperatingSystem osInfo = Environment.OSVersion;
+                return (osInfo.Platform.ToString());
```

Comment:
This wont work. [`System.Platformid`](https://docs.microsoft.com/en-us/dotnet/api/system.platformid?view=netcore-2.0) does not contain commonly accepted values for the platform part of a `User-Agent` header. For example, macOS should be `Macintosh` and all (recent) Windows should be `Windows NT`, and most Linux should be `Linux` or `X11`.